ddp-workspace/projects/ddp-dsm-ui/src/app/services/auth.service.ts

```diff
@@ -89,11 +89,11 @@ export class Auth {
 
   constructor(private router: Router, private activatedRoute: ActivatedRoute, private http: HttpClient,
               private sessionService: SessionService, private role: RoleService,
-              private compService: ComponentService, private dsmService: DSMService, private localStorageService: LocalStorageService
+              private compService: ComponentService, private dsmService: DSMService, private localStorageService: SessionStorageService
                ) {
     // Add callback for lock `authenticated` event
     this.lock.on('authenticated', (authResult: any) => {
-      localStorage.setItem(Auth.AUTH0_TOKEN_NAME, authResult.idToken);
+      sessionStorage.setItem(Auth.AUTH0_TOKEN_NAME, authResult.idToken);
```

Comment:
Why not keep the auth0 token in the `localStorage` and the study in the `sessionStorage`, and change the authentication guards and log in page?
This way the user can log in once, use separate tabs, and once they log out in one tab, they are out of DSM. This way the log in flow remains unchanged to the user

Reply:
Great question and one I thought about while doing this.  It's a similar issue to storing the realm in local storage.  If the user logs in with one set of credentials in the first tab and a different set of credentials in a future tab, the code will incorrectly attribute any follow-on work in the first tab to the user of the second tab.

Reply:
We may also want to significantly refactor our Auth0 work based on the Single Page Application (SPA) advice here; https://auth0.com/docs/secure/security-guidance/data-security/token-storage and implementation examples here: https://auth0.com/docs/libraries/auth0-single-page-app-sdk